tensorboard/webapp/metrics/effects/index.ts

```diff
@@ -261,6 +263,23 @@ export class MetricsEffects implements OnInitEffects {
     })
   );
 
+  private readonly addOrRemovePin$ = this.actions$.pipe(
+    ofType(actions.cardPinStateToggled),
+    withLatestFrom(this.getVisibleCardFetchInfos()),
```

Comment:
(For a followup PR): 

Can we check getShouldPersistSettings() before deciding whether to write state?

https://github.com/tensorflow/tensorboard/blob/04a064fd002b3989fc0d3558db6658986d55d4d0/tensorboard/webapp/persistent_settings/_redux/persistent_settings_selectors.ts#L26

The idea is:
* Internally we have a share feature (in tbcorp).
* If you load a shared tensorboard, we don't believe any decisions you make should be persisted in local storage and affect your regular usage. 
* getShouldPersistSettings captures this (it returns true by default but returns false if you load a shared dashboard)

Reply:
IIUC, if a shared TensorBoard is used, we should not **store** user activities in local storage. To determine this, we need to check if  `getShouldPersistSettings` is true. If so, we can store in the local storage; otherwise, we should not.  Is this correct?

Reply:
Yes, that is exactly what I would recommend.